### PR TITLE
feat: added select all button to customize columns modal; added styles; revised instruction text

### PR DESCRIPTION
**Pull request by kai-liu**
Contributes to #1613, #1981

Implemented select all functionalities to the columns. 
Implemented style changes to match the style of a Datagrid table.
Revised text to improve clarity.

#### What did you change?
within the `Datagrid` directory:

`DraggableElement.js` Added `selected` attribute to help with css selectors
`addons/CustomizeColumns/Actions.js` Removed search button, as required by design. (the button has only been commented out, in case future design changes requires the button again)
`CustomizeColumns/Columns.js` Added select-all element and implement logic to support it
`CustomizeColumns/CustomizeColumnsModal.js` Added logic to show number of visible columns vs total number of columns
`/styles/addons/CustomizeColumnsModal.scss` Style changes
`/styles/draggableElement.scss' Style changes

#### How did you test and verify your work?
storybook

### CODE REVIEW

File: packages/cloud-cognitive/src/components/Datagrid/Datagrid/DraggableElement.js
Comment (markkulube) [inline packages/cloud-cognitive/src/components/Datagrid/Datagrid/DraggableElement.js:15]: if the  `TableSelectRow` no longer used this line should be deleted.

(the less comments and less commented out code, the easier it is to parse the module)

File: packages/cloud-cognitive/src/components/Datagrid/Datagrid/DraggableElement.js
Comment (markkulube) [inline packages/cloud-cognitive/src/components/Datagrid/Datagrid/DraggableElement.js:138]: If this handler is no longer used it should removed.

(the less comments and less commented out code, the easier it is to parse the module)

File: packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/Actions.js
Comment (markkulube) [inline packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/Actions.js:25]: if these arguments are no longer used they should be removed

(the less comments and less commented out code, the easier it is to parse the module)

File: packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/Actions.js
Comment (markkulube) [inline packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/Actions.js:57]: commented out code should really be cleaned out (there are of course exceptions).

File: packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/Actions.js
Comment (markkulube) [inline packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/Actions.js:68]: commented out code should really be cleaned out (there are of course exceptions).

File: packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/Columns.js
Comment (markkulube) [inline packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/Columns.js:22]: commented out code should really be cleaned out (there are of course exceptions).

File: packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/Columns.js
Comment (markkulube) [inline packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/Columns.js:39]: commented out code should really be cleaned out (there are of course exceptions).

File: packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/Columns.js
Comment (markkulube) [inline packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/Columns.js:61]: commented out code should really be cleaned out (there are of course exceptions).

**Review by markkulube — CHANGES_REQUESTED** [review]
@kai-liu 

It seems like there is a lot more _commented out code_ than one would except. Yes, there are exceptional situations wherein _lines of commented out code_ are kept in production deployments. Though nothing comes immediately because I generally rely on version history should there be a need to restore a previous code snippet. 

My guess is the commented out code is being left in case future design requirements warrants a revert. I would let the developer of tomorrow worry about code needed for future enhancements and/or feature functions.

There is no absolute rule of thumb for commenting, but generally comments  should reserved to explain code patterns aren't easily understandable or the reason for doing something unconventional.

Ideally though, code should be self documenting, i.e. to write code that is self explanatory.

I will leave it to the maintainers to override this change request should the commenting style be compliant with repo standards.

Otherwise before moving forward to other refactor I would start by removing all unnecessary comments and/or dead code!

cc: @matthewgallo @elycheea @hoiwingh

File: packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/Columns.js
Comment (kai-liu) [reply packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/Columns.js:35]: This code is not commented out and therefore should not be removed.

File: packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/Columns.js
Comment (markkulube) [inline packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/Columns.js:72]: Perhaps try using array functions. This could end up looking something like

```
list.forEach((li, i) => {
       if(!columns[i].id.includes(id)) continue
       li[i].setAttribute('selected', isVisible)
})
```

File: packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/Columns.js
Comment (markkulube) [inline packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/Columns.js:80]: @kai-liu 

A `forEach `loop could be used here as well.

File: packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/Columns.js
Comment (markkulube) [inline packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/Columns.js:86]: @kai-liu 

You could appeal to the tenary operator and reduce to one line:
```
return return `${blockClass}__draggable-handleHolder${getVisibleColumnsCount() > 0 ? ' -selected'} : ''`;
```

File: packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/Columns.js
Comment (markkulube) [inline packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/Columns.js:95]: @kai-liu 

`if-else` block should be reducible to a one liner

```
target[0].setAttribute('selected', JSON.stringify(getVisibleColumnsCount() > 0))
```

File: packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/Columns.js
Comment (markkulube) [inline packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/Columns.js:117]: @kai-liu 

There is duplicate code here. So I would try to abstract out what's repeated.

File: packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/CustomizeColumnsModal.js
Comment (markkulube) [inline packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/CustomizeColumnsModal.js:90]: @kai-liu 

Probably another opportunity to use array functions / forEach loop.

**Review by markkulube — CHANGES_REQUESTED** [review]
Hello @kai-liu 

Overall looks good.

There some opportunities for to refactor` for-loop`s into `forEach` loops. Same thing but in the Javascript world we love array functions where they can be used. They tend to simplify the code (`forEach`, `filter`, `reduce`, `map`).
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array#instance_methods

If you do decide to transition into array functions,  do so in a granular fashion. This will make it easier to ensure changes to each snippet does not result in a functional regression.

Should you deem that is unnecessary and/or better to leave as is, that's okay too. Just let respond back to the change request with your rationale.

cc: @hoiwingh @matthewgallo @markkulube

**Review by holmansze — COMMENTED** [review]
@kai-liu thanks for the enhancement. A few comments:
1. The convention is to use single quotes for string literals and package names, double quotes for React props or element attributes. Are the changes made by prettifier? Can you revert to single quotes (follow convention of existing code) and then do `npm run lint` to verify?
2. Why are we taking out "reset to default"? I think the request is to add a select all / unselect all checkbox in the customize column modal. It doesn't seem to mention that we need to remove the "reset to default" feature. Is this an explicit request from design?
3. Can we further simplify the select / unselect all logic? On a high level, when the "select/unselect all" checkbox is clicked, there should be a for-loop that selects / unselects each column checkbox by calling the corresponding on change callback. The existing on change callback doesn't need to be changed. Also, we can probably implement a function which walks the column checkbox states and return a value that indicates if all/some/none column checkboxes are checked. This can be used to populate the select/unselect all checkbox state.

Please ping me if you need clarification and we can discuss in details.

**Pull request closed without merging**
